analyses/chromosomal-instability/02b-plot-chr-instability-by-histology.Rmd

```diff
@@ -0,0 +1,195 @@
+```
+
+### Read in data 
+
+Set up metadata
+
+```{r}
+# Read in the metadata
+metadata <- readr::read_tsv(metadata_file)
+```
+
+Load in the localized by histology breakpoint data. 
+
+```{r}
+hist_break_local <- readr::read_rds(hist_binned_breaks_file)
+```
+
+Read in the total intersection breaks densities TSV that was made by the 
+`00-setup-breakpoint-data.R` script. 
+
+```{r}
+total_break_densities_df <- readr::read_tsv(
+  total_densities_file
+) %>%
+  # Add on the short histology column from the metadata for plotting purposes
+  dplyr::inner_join(metadata %>% dplyr::select(Kids_First_Biospecimen_ID, short_histology),
+    by = c("samples" = "Kids_First_Biospecimen_ID")
+  )
+```
+
+## Plot CDF plot by histology
+
+```{r}
+breaks_cdf <- total_break_densities_df  %>%
+  # Only plot histologies groups with more than `min_samples` number of samples
+  dplyr::group_by(short_histology, add = TRUE) %>%
+  dplyr::filter(dplyr::n() > params$min_samples) %>%
+  # Calculate histology group mean
+  dplyr::mutate(
+    hist_mean = mean(breaks_density),
+    hist_rank = rank(breaks_density) / dplyr::n()
+  ) %>%
+  dplyr::ungroup() %>%
+  # Now we will plot these as cummulative distribution plots
+  ggplot2::ggplot(ggplot2::aes(
+    x = hist_rank, 
+    y = breaks_density,
+    color = short_histology
+  )) +
+  ggplot2::geom_point() +
+  # Add summary line for mean
+  ggplot2::geom_hline(ggplot2::aes(yintercept = hist_mean)) +
+  # Separate by histology
+  ggplot2::facet_wrap(~short_histology, nrow = 1) +
+  ggplot2::theme_classic() +
+  ggplot2::xlab("") +
+  ggplot2::ylab("Chromosomal Breaks per Mb") +
+  # Transform to log10 make non-log y-axis labels
+  ggplot2::scale_y_continuous(trans = "log10", breaks = c(0, 1, 10, 30)) +
+  ggplot2::theme(legend.position = "none") +
+  # Get rid of x-axis ticks
+  ggplot2::theme(
+    axis.text.x = ggplot2::element_blank(),
+    axis.ticks.x = ggplot2::element_blank(),
+    strip.text = ggplot2::element_text(size = 10)
+  )
+
+# Save the plot to a png
+ggplot2::ggsave(cdf_plot_file,
+  plot = breaks_cdf, width = 40, height = 20, unit = "cm"
+)
+```
+
+Print from png so rendering is smoother.
+![Breaks Density CDF Plot](./plots/breaks_cdf_plot.png)
+
+### Plot the breakpoints for each tumor-type
+
+Here we will plot median number of break points for the tumor-type group per 
+each bin.
+
+```{r}
+purrr::imap(hist_break_local, function(.x, name = .y) {
+  # Make the combo plot
+  multipanel_break_plot(
+    granges_list = .x,
+    plot_name = name,
+    y_val = "total_counts",
+    y_lab = "Total Breaks per Mb",
+    plot_dir = hist_plots_dir
+  )
+})
+```
+
+Zip up the PNG files into one file. 
+
+```{r}
+# Declare name of zip file
+zip_file <- paste0(hist_plots_dir, ".zip")
+
+# Remove any current zip_file of this name so we can overwrite it
+if (file.exists(zip_file)) {
+  file.remove(zip_file)
+}
+# Zip up the plots
+zip(zip_file, hist_plots_dir)
```

Comment:
When you make the zip file this way, they get put in plots/tumor-type within the folder, which is not really what we want.

Instead, you can try something like: 
```r
gzfile = paste0(hist_plots_dir, ".tar.gz")
plot_files = list.files(hist_plots_dir, full.names = TRUE, pattern = ".png")
tar(gzfile, plot_files, compression = "gzip")
```

Reply:
For future reference, this didn't work^.